src/cdomains/apron/apronDomain.apron.ml

```diff
@@ -768,71 +815,102 @@ struct
     dprintf "%s: %a not leq %a" (name ()) pretty x pretty y
 end
 
-module D2 = DWithOps (DHetero)
-
+module type S2 =
+sig
+  module Man: Manager
+  module Tracked : Tracked
+  module Bounds : module type of Bounds (Man)
+  include module type of AOps (Tracked) (Man)
+  include Tracked
+  include SLattice with type t = Man.mt A.t
+
+  type 'a aproncomponents_t = { apr : t; priv : 'a; }
+
+  val exp_is_cons : exp -> bool
+  val assert_cons : t -> exp -> bool -> t
+  val assert_inv : t -> exp -> bool -> t
+  val check_assert : t -> exp -> [> `False | `Top | `True ]
+  val eval_interval_expr : t -> exp -> Z.t option * Z.t option
+  val eval_int : t -> exp -> IntDomain.IntDomTuple.t
+
+  val equal_aproncomponents_t :
+    ('a -> 'a -> bool) ->
+    'a aproncomponents_t -> 'a aproncomponents_t -> bool
+  val compare_aproncomponents_t :
+    ('a -> 'a -> int) ->
+    'a aproncomponents_t -> 'a aproncomponents_t -> int
+  val aproncomponents_t_to_yojson :
+    ('a -> Printable.json) -> 'a aproncomponents_t -> Printable.json
```

Comment:
It would be good to get `aproncomponents_t` out of here, because it's an awkward place for these. If it's not possible, then I think it's also possible to use `[@@deriving ...]` inside the signature as well.

Reply:
Now the deriving is done for the type outside the signature.